std/fmt/index.zig

```diff
@@ -152,7 +154,7 @@ pub fn formatType(
                     if (info.child == u8) {
                         return formatText(value, fmt, context, Errors, output);
                     }
-                    return format(context, Errors, output, "{}@{x}", @typeName(T.Child), @ptrToInt(value));
+                    @compileError("Unexpected path 1 for possibly printing a pointer");
```

Comment:
this is supposed to work. why is this now a compile error?

Reply:
Added a test that exercises that path and changed the code.